pkg/kubelet/network/hostport/hostport_syncer.go

```diff
@@ -87,35 +104,31 @@ func (hp *hostport) String() string {
 }
 
 //openPodHostports opens all hostport for pod and returns the map of hostport and socket
-func (h *handler) openHostports(pod *v1.Pod) error {
+func (h *hostportSyncer) openHostports(podHostportMapping *PodPortMapping) error {
 	var retErr error
 	ports := make(map[hostport]closeable)
-	for _, container := range pod.Spec.Containers {
-		for _, port := range container.Ports {
-			if port.HostPort <= 0 {
-				// Ignore
-				continue
-			}
-			hp := hostport{
-				port:     port.HostPort,
-				protocol: strings.ToLower(string(port.Protocol)),
-			}
-			socket, err := h.portOpener(&hp)
-			if err != nil {
-				retErr = fmt.Errorf("Cannot open hostport %d for pod %s: %v", port.HostPort, kubecontainer.GetPodFullName(pod), err)
-				break
-			}
-			ports[hp] = socket
+	for _, port := range podHostportMapping.PortMappings {
+		if port.HostPort <= 0 {
+			// Ignore
```

Comment:
Please log this. should validation already prevent it? 
Also suggest a better comment.